cfgov/core/utils.py

```diff
@@ -24,18 +24,26 @@
 
 LINK_ICON_TEXT_CLASSES = ['a-link_text']
 
-# Regular expression to match <a> links in HTML strings
-A_TAG = re.compile(
-    # Match an <a containing any attributes
-    r'<a [^>]*?>'
-    # And match everything inside before the closing </a>
-    r'.+?(?=</a>)'
-    # Then match the closing </a>
-    r'</a>'
+# Regular expression format string that will match any tag <tag_name> and group
+# its contents.
+TAG_RE = (
+    # Match an <tag_name[ attributes]>. If tag_name is not followed by a space
+    # and any characters except >, it must be followed by >.
```

Comment:
See my reply below explaining the pattern. `<{tag_name}.*>` matches `<aside` in our tests (and then closes at the first `</a>` it finds, and replaces the contents of the `<aside>` up to the `</a>` with `<a ...></a>`, which makes the HTML invalid).

I'm not sure I understand what your concern is... could you give an example?

Reply:
I couldn't quickly find a real example, but say we had some template code like this:

```jinja
<p {% if some_condition %}class="some-class"{% endif %}>
```

If `some_condition` is falsy, then we end up with this output:

```html
<p >
```

which I think would not be matched by your regex, because it requires a non-`>` character following the space.

I see your point about errantly matching `aside`. Maybe the fix, then, is to make that post-`tag_name` group match a space plus _zero_ or more additional characters?

Reply:
Makes sense. I'll add a test against that case too.

Reply:
I've done that in 8504a16, I've also set the space to `\s+` so that any amount of space will match, not just a single whitespace character.